File: dohyeon/Programmers/Level_2/Lv2_예상_대진표.py
import math
def solution(n,a,b):
    answer = 0
    a, b = min(a, b), max(a, b)
    round = math.log(n) / math.log(2)
    left = 1
    right = n
    
    while True :
        mid = math.floor((left + right) / 2)
        if((a >= left and a <= mid) and (b <= right and b > mid)) :
            return round
        if((a >= left and a <= mid) and (b >= left and b <= mid)) :
            right = mid
        else :
            left = mid
        round = round - 1
    return answer

File: seongwon/Programmers/Lv2_예상_대진표.cpp
#include <iostream>
using namespace std;

int solution(int n, int a, int b)
{
    int answer = 0;
    while(a!=b){    //a==b==1일때 만남
        a = (a+1) >> 1; // 홀수일 수 있으니 +1 후 나누기 2
        b = (b+1) >> 1;
        answer++;
    }
    return answer;
}
